packages/nx/src/plugins/js/lock-file/utils/pnpm-normalizer.ts

```diff
@@ -20,8 +22,38 @@ const LOCKFILE_YAML_FORMAT = {
   sortKeys: false,
 };
 
+const ROOT_KEYS_ORDER = {
+  lockfileVersion: 1,
+  // only and never are conflict options.
+  neverBuiltDependencies: 2,
+  onlyBuiltDependencies: 2,
+  overrides: 3,
+  packageExtensionsChecksum: 4,
+  patchedDependencies: 5,
+  specifiers: 10,
+  dependencies: 11,
+  optionalDependencies: 12,
+  devDependencies: 13,
+  dependenciesMeta: 14,
+  importers: 15,
+  packages: 16,
+};
+
+function isV6Lockfile(data: InlineSpecifiersLockfile | Lockfile) {
+  return data.lockfileVersion.toString().startsWith('6.');
```

Comment:
Is this a semver? Could be `semver.major(data.lockfileVersion) === 6`

Reply:
`semver.major` wraps version string in a 200-line large class so that would be quite an overkill for such a simple usage. Plus this is what `pnpm` original codebase is using so it's closer to the source.